matrix/Dcm.hpp

```diff
@@ -28,20 +40,69 @@ class Dcm : public Matrix<Type, 3, 3>
 
     typedef Matrix<Type, 3, 1> Vector3;
 
+    /**
+     * Standard constructor
+     *
+     * Initializes to identity
+     */
     Dcm() : Matrix<Type, 3, 3>()
     {
         (*this) = eye<Type, 3>();
     }
 
+    /**
+     * Constructor from array
+     *
+     * @param _data pointer to array
+     */
     Dcm(const Type *data_) : Matrix<Type, 3, 3>(data_)
     {
     }
 
+    /**
+     * Copy constructor
+     *
+     * @param other Matrix33 to set dcm to
+     */
     Dcm(const Matrix<Type, 3, 3> &other) : Matrix<Type, 3, 3>(other)
     {
     }
 
+    /**
+     * Constructor from quaternion
+     *
+     * Instance is initialized from quaternion representing
+     * coordinate transformation from frame 2 to frame 1.
+     *
+     * @param q quaternion to set dcm to
+     */
     Dcm(const Quaternion<Type> & q) {
+        set_from_quaternion(q);
+    }
+
+    /**
+     * Constructor from euler angles
+     *
+     * This sets the transformation matrix from frame 2 to frame 1 where the rotation
+     * from frame 1 to frame 2 is described by a 3-2-1 intrinsic Tait-Bryan rotation sequence.
+     *
+     *
+     * @param euler euler angle instance
+     */
+    Dcm(const Euler<Type> & euler) {
+        set_from_euler(euler);
+    }
+
+
+    /**
+     * Set from quaternion
+     *
+     * Instance is set from quaternion representing
+     * transformation from frame 2 to frame 1.
```

Comment:
I agree that it should be from 1 to 2. Thanks for pointing out the mistake I did in the initial description.